lib/carelink/bolusNWizard.js

```diff
@@ -470,129 +472,164 @@ module.exports = function(timezone, model) {
   }
 
   var datumsBySeqNum = null, findNormal = null, findSquare = null, findWizard = null;
+  var pointer = 0, cachedObjects = null;
```

Comment:
Your "buildCache" and cache manipulation routines don't touch "cachedObjects" -- but instead use "datumsBySeqNum". The "cachedObjects" is more of your result list. That's pretty confusing. Can you adjust the names of things so that they agree with each other, please?